memor/prompt.py

```diff
@@ -287,21 +287,29 @@ def from_json(self, json_object: Union[str, Dict[str, Any]]) -> None:
             role = Role(loaded_obj["role"])
             template = PresetPromptTemplate.DEFAULT.value
             if "template" in loaded_obj:
-                template_obj = PromptTemplate()
-                template_obj.from_json(loaded_obj["template"])
-                template = template_obj
+                template = PromptTemplate()
+                template.from_json(loaded_obj["template"])
             memor_version = loaded_obj["memor_version"]
             date_created = datetime.datetime.strptime(loaded_obj["date_created"], DATE_TIME_FORMAT)
             date_modified = datetime.datetime.strptime(loaded_obj["date_modified"], DATE_TIME_FORMAT)
             selected_response_index = loaded_obj["selected_response_index"]
         except Exception:
             raise MemorValidationError(INVALID_PROMPT_STRUCTURE_MESSAGE)
-        if len(responses) > 0:
-            self._selected_response = responses[selected_response_index]
-            self._selected_response_index = selected_response_index
-        else:
-            self._selected_response = None
-            self._selected_response_index = 0
+        _validate_string(message, "message")
+        if tokens:
+            _validate_pos_int(tokens, "tokens")
+        _validate_message_id(_id)
+        _validate_string(memor_version, "memor_version")
+        return message, tokens, _id, responses, role, template, memor_version, date_created, date_modified, selected_response_index
```

Comment:
The `Prompt` class has many parameters, so returning them in this format is not a good approach. I suggest returning a dictionary instead.

Reply:
fixed in [9523075](https://github.com/openscilab/memor/pull/120/commits/9523075ef2ebd4a782024e43d9cd231834bd51a5)